dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/ConnectorArguments.java

```diff
@@ -366,6 +368,21 @@ public class ConnectorArguments extends DefaultArguments {
           .ofType(String.class)
           .describedAs("principal/host");
 
+  private final OptionSpec<String> optionHiveSaslQop =
+      parser
+          .accepts(
+              OPT_HIVE_SASL_QOP,
+              "The QOP of the SASL connection between hadoop and the dumper. "
+                  + "Please check the 'hadoop.rpc.protection' configuration for your cluster. "
+                  + "Mapping between the 'hadoop.rpc.protection' value and SASL QOP:\n"
+                  + "authentication: auth\n"
+                  + "integrity: auth-int\n"
+                  + "privacy: auth-conf")
+          .withOptionalArg()
```

Comment:
Why is the argument optional?

Reply:
The user can either provide the option or not.
If they don't provide the option, the default is used (`auth-conf`).
If they do provide the option, it can either accept an argument or not. I think that in case of this option the argument should always be provided, i.e. it should be required, not optional.